Implemented Sawmill

LeviHarrison opened the pull request:
#57 this PR I've implemented Sawmill in almost all of the Functions, excluding `tokengen` and `updatealgolia` because those are weird. I've also added a spot in `config.yaml` for the credentials. I plan to add the Emoji improvements to Sawmill in another PR, just wanted to get this out.

LeviHarrison commented:
Sorry about all the cryptic commit names, I accidentally left in some things in that shouldn't be in production.

adilanchian commented on cmd/appleauth/appleauth.go:
```diff
@@ -19,6 +20,7 @@ type appleDevTokenResp struct {
 }
 
 var firestoreClient *firestore.Client
+var logger sawmill.Logger
```
🔥  sawmill.Logger 🔥 

LETS GO.

adilanchian commented on cmd/appleauth/appleauth.go:
```diff
@@ -35,7 +37,7 @@ func AuthorizeWithApple(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("[AuthorizeWithApple] [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnv", nil)
```
So I think in the `LogErr` method in `sawmill`, we should change the param order to be:

`operation, error, request`

Thoughts on this? I feel like there is a lot more readability in this type of layout.

adilanchian replied:
[nit]

`initWithEnv` -> `InitWithEnv`

LeviHarrison replied:
Sure! If that looks order looks better to you I'm glad to implement it.

adilanchian commented on cmd/appleauth/appleauth.go:
```diff
@@ -44,14 +46,14 @@ func AuthorizeWithApple(writer http.ResponseWriter, request *http.Request) {
 	if appleDevTokenReqErr != nil {
 		http.Error(writer, appleDevTokenReqErr.Error(), http.StatusInternalServerError)
-		log.Println(appleDevTokenReqErr.Error())
+		logger.LogErr(appleDevTokenReqErr, "appleDevToken Request", appleDevTokenReq)
```
[nit]

`appleDevToken ` -> `AppleDevToken `

adilanchian commented on cmd/appleauth/appleauth.go:
```diff
@@ -44,14 +46,14 @@ func AuthorizeWithApple(writer http.ResponseWriter, request *http.Request) {
 	if appleDevTokenRespErr != nil {
 		http.Error(writer, appleDevTokenRespErr.Error(), http.StatusInternalServerError)
-		log.Println(appleDevTokenRespErr.Error())
+		logger.LogErr(appleDevTokenRespErr, "appleDevToken Response", nil)
```
[nit]

`appleDevToken` -> `AppleDevToken`

adilanchian commented on cmd/appleauth/appleauth.go:
```diff
@@ -60,7 +62,7 @@ func AuthorizeWithApple(writer http.ResponseWriter, request *http.Request) {
 	if appleDevTokenDecodeErr != nil {
 		http.Error(writer, appleDevTokenDecodeErr.Error(), http.StatusInternalServerError)
-		log.Printf("AppleAuth [appleDevToken Decoder]: %v", appleDevTokenDecodeErr)
+		logger.LogErr(appleDevTokenDecodeErr, "appleDevToken Decoder", nil)
```
[nit]

`appleDevToken` -> `AppleDevToken`

adilanchian commented on cmd/appleauth/appleauth.go:
```diff
@@ -71,7 +73,7 @@ func AuthorizeWithApple(writer http.ResponseWriter, request *http.Request) {
 	if renderErr != nil {
 		http.Error(writer, renderErr.Error(), http.StatusInternalServerError)
-		log.Printf("[AuthorizeWithApple] [render]: %v", renderErr)
+		logger.LogErr(renderErr, "render", nil)
```
[nit]

`render` -> `Render`

adilanchian commented on cmd/createappledevtoken/createappledevtoken.go:
```diff
@@ -31,15 +33,15 @@ func CreateAppleDevToken(writer http.ResponseWriter, request *http.Request) {
 	if devTokenErr != nil {
 		http.Error(writer, devTokenErr.Error(), http.StatusInternalServerError)
-		log.Printf("[CreateAppleDevToken] [getAppleDevToken]: %v", devTokenErr)
+		logger.LogErr(devTokenErr, "getAppleDevToken", nil)
```
[nit]

`getAppleDevToken` -> `GetAppleDevToken`

adilanchian commented on cmd/createappledevtoken/createappledevtoken.go:
```diff
@@ -109,6 +111,13 @@ func initWithEnv() error {
+		log.Printf("CreateAppleDevToken [Init Sawmill]: %v", err)
+	}
+
+	logger = sawmillLogger
 	firestoreClient = client
```
[nit]

In the other function you set `logger = sawmillLogger` under the `firestoreClient`. Can you change that here?

adilanchian commented on cmd/createprovideruser/createprovideruser.go:
```diff
@@ -23,14 +24,15 @@ type providerUser struct {
 	if initErr != nil {
 		http.Error(writer, initErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateProviderUser [Init failure]: %v", initErr)
+		logger.LogErr(initErr, "initWithEnv", nil)
```
[nit]

`initWithEnv` -> `InitWithEnv`

adilanchian commented on cmd/createprovideruser/createprovideruser.go:
```diff
@@ -40,7 +42,7 @@ func CreateProviderUser(writer http.ResponseWriter, request *http.Request) {
 	if reqDataErr != nil {
 		http.Error(writer, reqDataErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateProviderUser [reqData Decoder]: %v", reqDataErr)
+		logger.LogErr(reqDataErr, "reqData Decoder", request)
```
[nit]

`reqData ` -> `ReqData `

adilanchian commented on cmd/createprovideruser/createprovideruser.go:
```diff
@@ -57,7 +59,7 @@ func CreateProviderUser(writer http.ResponseWriter, request *http.Request) {
 	if writeErr != nil {
 		http.Error(writer, writeErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateProviderUser [fireStore Set]: %v", writeErr)
+		logger.LogErr(writeErr, "fireStore Set", request)
```
[nit]

`fireStore -> FireStore`

adilanchian commented on cmd/createsocialplaylist/createsocialplaylist.go:
```diff
@@ -64,7 +66,7 @@ func CreateSocialPlaylist(writer http.ResponseWriter, request *http.Request) {
 	if err != nil {
 		http.Error(writer, err.Error(), http.StatusInternalServerError)
-		log.Printf("CreateSocialPlaylist [initWithEnv]: %v", err.Error())
+		logger.LogErr(err, "initWithEnv", nil)
```
[nit]

`initWithEnv` -> `InitWithEnv`

adilanchian commented on cmd/createsocialplaylist/createsocialplaylist.go:
```diff
@@ -74,7 +76,7 @@ func CreateSocialPlaylist(writer http.ResponseWriter, request *http.Request) {
 	if jsonDecodeErr != nil {
 		http.Error(writer, jsonDecodeErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateSocialPlaylist [socialPlaylistReq Decoder]: %v", jsonDecodeErr)
+		logger.LogErr(jsonDecodeErr, "socialPlaylistReq Decoder", request)
```
[nit]

`socialPlaylistReq ` -> `SocialPlaylistReq `

adilanchian commented on cmd/createsocialplaylist/createsocialplaylist.go:
```diff
@@ -92,7 +94,7 @@ func CreateSocialPlaylist(writer http.ResponseWriter, request *http.Request) {
 		if socialRefreshTokenErr != nil {
 			http.Error(writer, socialRefreshTokenErr.Error(), http.StatusBadRequest)
-			log.Printf("CreateSocialPlaylist [refreshToken]: %v", socialRefreshTokenErr)
+			logger.LogErr(socialRefreshTokenErr, "refreshToken", request)
```
[nit]

`refreshToken` -> `RefreshToken`

adilanchian commented on cmd/createsocialplaylist/createsocialplaylist.go:
```diff
@@ -127,7 +129,7 @@ func CreateSocialPlaylist(writer http.ResponseWriter, request *http.Request) {
 	if createReqErr != nil {
 		http.Error(writer, createReqErr.Error(), http.StatusBadRequest)
-		log.Printf("CreateSocialPlaylist [createPlaylist]: %v", createReqErr.Error())
+		logger.LogErr(createReqErr, "createPlaylist", request)
```
[nit]

`createPlaylist ` -> `CreatePlaylist `

adilanchian commented on cmd/createuser/createuser.go:
```diff
@@ -25,7 +27,7 @@ func CreateUser(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateUser [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnv", nil)
```
[nit]

`initWithEnv ` -> `InitWithEnv `

adilanchian commented on cmd/createuser/createuser.go:
```diff
@@ -34,15 +36,15 @@ func CreateUser(writer http.ResponseWriter, request *http.Request) {
 	if jsonDecodeErr != nil {
 		http.Error(writer, jsonDecodeErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateUser [social.CreateUserReq Decoder]: %v", jsonDecodeErr)
+		logger.LogErr(jsonDecodeErr, "social.CreateUserReq Decoder", request)
```
[nit]

`social.CreateUserReq Decoder` -> `CreateUserReq Decoder`

adilanchian commented on cmd/createuser/createuser.go:
```diff
@@ -34,15 +36,15 @@ func CreateUser(writer http.ResponseWriter, request *http.Request) {
 	if socialPlatDocRef == nil {
 		http.Error(writer, jsonDecodeErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateUser [social.CreateUserReq Decoder]: %v", jsonDecodeErr)
+		logger.LogErr(jsonDecodeErr, "social.CreateUserReq Decoder", request)
```
[nit]

`social.CreateUserReq Decoder` -> `CreateUserReq Decoder`

adilanchian commented on cmd/createuser/createuser.go:
```diff
@@ -62,7 +64,7 @@ func CreateUser(writer http.ResponseWriter, request *http.Request) {
 	if writeErr != nil {
 		http.Error(writer, writeErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateUser [fireStore Set]: %v", writeErr)
+		logger.LogErr(writeErr, "fireStore Set", nil)
```
[nit]

`fireStore` -> `FireStore`

adilanchian commented on cmd/doesuserdocexist/doesuserdocexist.go:
```diff
@@ -36,7 +38,7 @@ func DoesUserDocExist(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("DoesUserDocExist [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnv", nil)
```
[nit]

`initWithEnv` -> `InitWithEnv`

adilanchian commented on cmd/doesuserdocexist/doesuserdocexist.go:
```diff
@@ -46,7 +48,7 @@ func DoesUserDocExist(writer http.ResponseWriter, request *http.Request) {
 	if reqDataErr != nil {
 		http.Error(writer, reqDataErr.Error(), http.StatusInternalServerError)
-		log.Printf("DoesUserDocExist [reqData Decoder]: %v", reqDataErr)
+		logger.LogErr(reqDataErr, "reqData Decoder", request)
```
[nit]

`reqData` -> `ReqData`

adilanchian commented on cmd/getspotifymedia/getspotifymedia.go:
```diff
@@ -104,7 +106,7 @@ func GetSpotifyMedia(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("GetSpotifyMedia [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnv", nil)
```
[nit]

`initiWithEnv` -> `InitWithEnv`

adilanchian commented on cmd/getspotifymedia/getspotifymedia.go:
```diff
@@ -113,15 +115,15 @@ func GetSpotifyMedia(writer http.ResponseWriter, request *http.Request) {
 	if mediaDataErr != nil {
 		http.Error(writer, mediaDataErr.Error(), http.StatusInternalServerError)
-		log.Printf("GetSpotifyMedia [getMediaFromFirestore]: %v", mediaDataErr)
+		logger.LogErr(mediaDataErr, "getMediaFromFirestore", request)
```
[nit]

`getMediaFromFirestore` -> `GetMediaFromFirestore`

adilanchian commented on cmd/getspotifymedia/getspotifymedia.go:
```diff
@@ -137,7 +139,7 @@ func GetSpotifyMedia(writer http.ResponseWriter, request *http.Request) {
 	if credErr != nil {
 		http.Error(writer, credErr.Error(), http.StatusInternalServerError)
-		log.Printf("GetSpotifyMedia [getCreds]: %v", credErr)
+		logger.LogErr(credErr, "getCreds", nil)
```
[nit]

`getCreds` -> `GetCreds`

adilanchian commented on cmd/getspotifymedia/getspotifymedia.go:
```diff
@@ -176,7 +178,7 @@ func GetSpotifyMedia(writer http.ResponseWriter, request *http.Request) {
 		if firestoreMediaDataErr != nil {
 			http.Error(writer, firestoreMediaDataErr.Error(), http.StatusInternalServerError)
-			log.Printf("GetSpotifyMedia [GetSpotifyPlaylist Switch]: %v", firestoreMediaDataErr)
+			logger.LogErr(firestoreMediaDataErr, "GetSpotifyPlaylist Switch", request)
```
[nit]

`GetSpotifyPlaylist Switch ` -> `GetSpotifyAlbum Switch `

adilanchian commented on cmd/socialplatform/socialplatform.go:
```diff
@@ -26,7 +28,7 @@ func CreateSocialPlatform(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("CreateSocialPlatform [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnvErr", nil)
```
[nit]

`initWithEnvErr` -> `InitWithEnv`

adilanchian commented on cmd/socialtokenrefresh/socialtokenrefresh.go:
```diff
@@ -49,7 +51,7 @@ func SocialTokenRefresh(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("SocialTokenRefresh [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnv", nil)
```
[nit]

`initWithEnv` -> `InitWithEnv`

adilanchian commented on cmd/socialtokenrefresh/socialtokenrefresh.go:
```diff
@@ -60,15 +62,15 @@ func SocialTokenRefresh(writer http.ResponseWriter, request *http.Request) {
 	if socialTokenErr != nil {
 		http.Error(writer, socialTokenErr.Error(), http.StatusInternalServerError)
-		log.Printf("SocialTokenRefresh [socialTokenReq Decoder]: %v", socialTokenErr)
+		logger.LogErr(socialTokenErr, "socialTokenReq Decoder", request)
```
[nit]

`socialTokenReq` -> `SocialTokenReq`

adilanchian commented on cmd/socialtokenrefresh/socialtokenrefresh.go:
```diff
@@ -60,15 +62,15 @@ func SocialTokenRefresh(writer http.ResponseWriter, request *http.Request) {
 	if platformErr != nil {
 		http.Error(writer, platformErr.Error(), http.StatusInternalServerError)
-		log.Printf("SocialTokenRefresh [getUserPlatforms]: %v", platformErr)
+		logger.LogErr(platformErr, "getUserPlatforms", request)
```
[nit]

`getUserPlatforms` -> `GetUserPlatforms`

adilanchian commented on cmd/socialtokenrefresh/socialtokenrefresh.go:
```diff
@@ -105,6 +107,13 @@ func initWithEnv() error {
+		log.Printf("SocialTokenRefresh [Init Sawmill]: %v", err)
+	}
+
+	logger = sawmillLogger
 	firestoreClient = client
```
You remove firestoreClient? Or am I reading this wrong?

LeviHarrison replied:
I think I just populated `sawmillClient` before `firestoreClient`. I'll fix that order.

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -48,7 +50,7 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("AuthorizeWithSpotify [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnvErr", nil)
```
[nit]

`initWithEnvErr` -> `initWithEnv`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -57,20 +59,20 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 	if authResponseErr != nil {
 		http.Error(writer, authResponseErr.Error(), http.StatusInternalServerError)
-		log.Printf("AuthorizeWithSpotify [spotifyAuthRequest Decoder]: %v", authResponseErr)
+		logger.LogErr(authResponseErr, "spotifyAuthRequest Decoder", request)
```
[nit]

`spotifyAuthRequest` -> `SpotifyAuthRequest`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -57,20 +59,20 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 	if spotifyMeReqErr != nil {
 		http.Error(writer, spotifyMeReqErr.Error(), http.StatusInternalServerError)
-		log.Printf("AuthorizeWithSpotify [http.NewRequest]: %v", spotifyMeReqErr)
+		logger.LogErr(spotifyMeReqErr, "http.NewRequest", spotifyMeReq)
```
[nit]

`http.NewRequest` -> `Request`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -79,7 +81,7 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 	if httpErr != nil {
 		http.Error(writer, httpErr.Error(), http.StatusBadRequest)
-		log.Printf("AuthorizeWithSpotify [client.Do]: %v", httpErr)
+		logger.LogErr(httpErr, "client.Do", spotifyMeReq)
```
[nit]

`client.Do` -> `GET Request`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -107,15 +109,15 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 	if userErr != nil {
 		http.Error(writer, userErr.Error(), http.StatusBadRequest)
-		log.Printf("AuthorizeWithSpotify: %v", userErr)
+		logger.LogErr(userErr, "getUser", nil)
```
[nit]

`getUser` ->`GetUser`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -125,7 +127,7 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 		if socialPlatErr != nil {
 			http.Error(writer, socialPlatErr.Error(), http.StatusBadRequest)
-			log.Printf("AuthorizeWithSpotify: [createSocialPlatform] %v", socialPlatErr)
+			logger.LogErr(socialPlatErr, "createSocialPlatform", nil)
```
[nit]

`createSocialPlatform` -> `CreateSocialPlatform`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -141,7 +143,7 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 		if customTokenErr != nil {
 			http.Error(writer, customTokenErr.Error(), http.StatusBadRequest)
-			log.Printf("AuthorizeWithSpotify [customToken]: %v", customTokenErr)
+			logger.LogErr(customTokenErr, "customToken", nil)
```
[nit]

`customToken` -> `CustomToken`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -168,7 +170,7 @@ func AuthorizeWithSpotify(writer http.ResponseWriter, request *http.Request) {
 		if customTokenErr != nil {
 			http.Error(writer, customTokenErr.Error(), http.StatusBadRequest)
-			log.Printf("AuthorizeWithSpotify [customToken]: %v", customTokenErr)
+			logger.LogErr(customTokenErr, "customToken", nil)
```
[nit]

`customToken` -> `CustomToken`

adilanchian commented on cmd/spotifyauth/spotifyauth.go:
```diff
@@ -201,9 +203,16 @@ func initWithEnv() error {
 	log.Println(currentProject)
 	log.Println(hostname)
 
+	logger = sawmillLogger
 	firestoreClient = client
```
Add `logger = sawmillLogger` after `firestoreClient = client`

adilanchian commented on cmd/usernameavailable/usernameavailable.go:
```diff
@@ -34,7 +36,7 @@ func UsernameAvailable(writer http.ResponseWriter, request *http.Request) {
 	if initWithEnvErr != nil {
 		http.Error(writer, initWithEnvErr.Error(), http.StatusInternalServerError)
-		log.Printf("DoesUserDocExist [initWithEnv]: %v", initWithEnvErr)
+		logger.LogErr(initWithEnvErr, "initWithEnv", nil)
```
[nit]

`initWithEnv` -> `InitWithEnv`

adilanchian commented on cmd/usernameavailable/usernameavailable.go:
```diff
@@ -44,7 +46,7 @@ func UsernameAvailable(writer http.ResponseWriter, request *http.Request) {
 	if reqDataErr != nil {
 		http.Error(writer, reqDataErr.Error(), http.StatusInternalServerError)
-		log.Printf("UsernameAvailable [reqData Decoder]: %v", reqDataErr)
+		logger.LogErr(reqDataErr, "reqData Decoder", request)
```
[nit]

`reqData` -> `ReqData`

adilanchian commented on internal/mock-config.yaml:
```diff
@@ -11,3 +11,4 @@ FIREBASE_PROJECTID_DEV: <AddRegisteredFirebaseProjectID>
 ALGOLIA_INDEX_NAME_DEV: <AddCreatedAlgoliaIndexName>
 APPLE_AUTH_TEMPLATE_PATH_DEV: cmd/appleauth/templates
 APPLE_PRIVATE_KEY_PATH_DEV: cmd/createappledevtoken/apple_private_key.p8
+GCLOUD_CONFIG: <ServiceAccountConfigJSONObjectPutItAllInOneLineAndWrapItWithSingleQuotes>
```
L.U.L.

Keep it for the memes kappa.

LeviHarrison commented:
Fixed up the capitalization issues, changed the order of the parameters for `LogErr`, and caught one very pesky typo. Regarding UpdateAlgolia, originally there were two ways to initialize a Sawmill Client, one which I used in most of the functions, and the other which I used in a few functions that didn't have `initWithEnv`. The problem was that the functions using the other way of initializing a Sawmill Client (including UpdateAlgolia), weren't working, so I assumed that it had to do with the functions. Later I found that way of initializing to be flawed, but I didn't put two and two together, so I still believed that UpdateAlgolia was just an anomaly.  I'll get around to adding Sawmill to it at some point, I just don't have my config stuff set up on this laptop right now so I can't test if it will load all the ENVs.
